### ISSUE TITLE
[Bug]: project depends on both chewie ^1.3.5 and share_plus ^7.0.1, version solving failed.

### ISSUE THREAD

**Issue by abdulmalekDery** [fluttercommunity/plus_plugins#1833]
### Platform

All

### Plugin

share_plus

### Version

7.0.1

### Flutter SDK

3.7.11

### Steps to reproduce

I'm trying to share_plus as dependency to my project but I'm getting the error that you will see in the logs

### Code Sample

```dart
cupertino_icons: ^1.0.2
  flutter_screenutil: ^5.5.4
  built_value: ^8.0.6
  built_collection: ^5.0.0
  wc_form_validators: ^1.0.0
  cached_network_image: ^3.2.1
  after_layout: ^1.1.0
  dio: ^4.0.6
  intl: ^0.17.0
  bloc: ^7.2.0
  flutter_bloc: ^7.3.3
  hydrated_bloc: ^7.1.0
  animations: ^2.0.3
  page_transition: ^2.0.2
  flutter_i18n: ^0.22.3
  flutter_translate_annotations: ^2.0.0
  dotted_border: ^2.0.0
  flutter_svg: ^0.23.0+1
  shimmer: ^2.0.0
  flutter_spinkit: ^5.1.0
  path_provider: ^2.0.9
  fl_country_code_picker: ^0.0.3+1
  date_time_picker: ^2.1.0
  pin_code_fields: ^7.4.0
  sliver_tools: ^0.2.8
  flutter_slidable: ^2.0.0
  table_calendar: ^3.0.8
  smooth_star_rating_null_safety: ^1.0.4+2
  dotted_line: ^3.1.0
  qr_flutter: ^4.0.0
  chewie: ^1.3.5
  video_player: ^2.4.7
  mobile_scanner: ^2.1.0
  google_maps_flutter: ^2.1.10
  flutter_switch: ^0.3.2
  image_picker: ^0.8.6+1
  carousel_slider: ^4.2.1
  story_view: ^0.14.0
  rich_text_controller: ^1.4.0
  geolocator: ^9.0.1
  uuid: ^3.0.7
  camera: ^0.10.3
  flutter_datetime_picker: ^1.5.1
  video_thumbnail: ^0.5.3
  pusher_channels_flutter: ^2.1.2
  crypto: ^3.0.2
  firebase_core: ^2.8.0
  firebase_messaging: ^14.3.0
  flutter_local_notifications: ^12.0.4
  hyperpay_plugin: ^2.0.0
  url_launcher: ^6.1.10
  permission_handler: ^10.2.0
  agora_rtc_engine: ^6.1.0
  agora_uikit: ^1.3.1
  screenshot: ^1.3.0
  tutorial_coach_mark: ^1.2.8
  shared_preferences: ^2.1.0
  searchfield: ^0.7.7
  pull_to_refresh: ^2.0.0
  share_plus: ^7.0.1
```

### Logs

```shell
Because no versions of wakelock_windows match >0.2.0 <0.2.1 and wakelock_windows >=0.2.1 depends on win32 ^3.0.0, wakelock_windows >0.2.0 requires win32 ^3.0.0.
And because wakelock_windows 0.2.0 depends on win32 ^2.0.0, wakelock_windows >=0.2.0 requires win32 ^2.0.0 or ^3.0.0.
And because share_plus >=7.0.1 depends on win32 >=4.0.0 <6.0.0 and wakelock 0.6.2 depends on wakelock_windows ^0.2.0, share_plus >=7.0.1 is incompatible with wakelock 0.6.2.
Because chewie >=1.3.5 depends on wakelock ^0.6.2 and no versions of wakelock match >0.6.2 <0.7.0, chewie >=1.3.5 requires wakelock 0.6.2.
Thus, share_plus >=7.0.1 is incompatible with chewie >=1.3.5.
So, because howwi depends on both chewie ^1.3.5 and share_plus ^7.0.1, version solving failed.
pub get failed
```

### Flutter Doctor

```shell
No need
```

### Checklist before submitting a bug

- [X] I Google'd a solution and I couldn't find it
- [X] I searched on StackOverflow for a solution and I couldn't find it
- [X] I read the README.md file of the plugin
- [X] I'm using the latest version of the plugin
- [X] All dependencies are up to date with `flutter pub upgrade`
- [X] I did a `flutter clean`
- [X] I tried running the example project

**Comment by vbuberen** [comment]
Duplicate of https://github.com/fluttercommunity/plus_plugins/issues/1812 

It is the problem of `wakelock` dependency, which is not updated for a long time. Read the referenced issue. 
EIther wait till `wakelock` and `chewie` are updated or use dependency_overrides.

**Comment by abdulmalekDery** [comment]
@vbuberen 
Do you have any idea how to use dependency_overrides in that case?

**Comment by vbuberen** [comment]
https://letmegooglethat.com/?q=Flutter+dependency+overrides
